zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -172,14 +143,95 @@ public boolean isValid(String id) {
     }
   }
 
+  /**
+   * Initialize a new ClientUriDomainMappingHelper instance if no one has been instantiated for the ACL provider.
+   * @param zks
+   */
   private ClientUriDomainMappingHelper getUriDomainMappingHelper(ZooKeeperServer zks) {
     if (uriDomainMappingHelper == null) {
       synchronized (this) {
         if (uriDomainMappingHelper == null) {
-          uriDomainMappingHelper = new ZkClientUriDomainMappingHelper(zks);
+          ZkClientUriDomainMappingHelper helper = new ZkClientUriDomainMappingHelper(zks);
+          // Set up AuthInfo updater to refresh connection AuthInfo on any client domain changes.
+          helper.setDomainAuthUpdater((cnxn, clientUriToDomainNames) -> {
```

Comment:
The return value of `helper.setDomainAuthUpdater` is not used here. Should we throw exception or do something if this returns false?